src/LibraryManager/Providers/Unpkg/UnpkgProvider.cs

```diff
@@ -1,98 +1,41 @@
-            // TODO: {alexgav} Do we need multiple instances of CacheService?
-            _cacheService = new CacheService(WebRequestHandler.Instance);
         }
 
-        public string Id => IdText;
-
-        public string NuGetPackageId { get; } = "Microsoft.Web.LibraryManager.Build";
-
-        public IHostInteraction HostInteraction { get; }
+        public override string Id => IdText;
 
-        private ILibraryNamingScheme LibraryNamingScheme { get; } = new VersionedLibraryNamingScheme();
-
-        public ILibraryCatalog GetCatalog()
+        public override ILibraryCatalog GetCatalog()
         {
+            // TODO: sort out the WebRequestHandler dependency
             return _catalog ?? (_catalog = new UnpkgCatalog(Id, LibraryNamingScheme, HostInteraction.Logger, WebRequestHandler.Instance, _infoFactory, _packageSearch));
         }
 
-        // TODO: {alexgav} Could got to a command provider base class
-        internal string CacheFolder
-        {
-            get { return Path.Combine(HostInteraction.CacheDirectory, Id); }
-        }
-
-        public string LibraryIdHintText => Resources.Text.UnpkgProviderHintText;
-
-        public bool SupportsLibraryVersions => true;
-
-        public async Task<ILibraryOperationResult> InstallAsync(ILibraryInstallationState desiredState, CancellationToken cancellationToken)
-        {
-            if (cancellationToken.IsCancellationRequested)
-            {
-                return LibraryOperationResult.FromCancelled(desiredState);
-            }
-
-            //Expand the files property if needed
-            ILibraryOperationResult updateResult = await UpdateStateAsync(desiredState, cancellationToken);
-            if (!updateResult.Success)
-            {
-                return updateResult;
-            }
-
-            desiredState = updateResult.InstallationState;
-
-            // Refresh cache if needed
-            ILibraryOperationResult cacheUpdateResult = await RefreshCacheAsync(desiredState, cancellationToken);
-            if (!cacheUpdateResult.Success)
-            {
-                return cacheUpdateResult;
-            }
-
-            // Check if Library is already up tp date
-            if (IsLibraryUpToDate(desiredState, cancellationToken))
-            {
-                return LibraryOperationResult.FromUpToDate(desiredState);
-            }
-
-            // Write files to destination
-            return await WriteToFilesAsync(desiredState, cancellationToken);
-        }
+        public override string LibraryIdHintText => Resources.Text.UnpkgProviderHintText;
 
         /// <summary>
         /// Returns the UnpkgLibrary's name.
         /// </summary>
         /// <param name="library"></param>
         /// <returns></returns>
-        public string GetSuggestedDestination(ILibrary library)
+        public override string GetSuggestedDestination(ILibrary library)
         {
             if (library != null && library is UnpkgLibrary unpkgLibrary)
```

Comment:
Can this ever get called on UpkgProvider with non-UnpkgLibrary? Same for other providers? I would guess it shouldn't, in which case we cane use use library.Name instead of casting to unpkgLibrary and move this method to base provider as well, no? cdnjs, jsdeliver and unpkg all do the same thing. File system could override.

Reply:
Technically it would probably be fine, but I don't see why giving up the type check is critical.  This is not on a hot path, and if anything ought to change, it feels like we should be throwing when the `library is ProviderLibrary` check fails (assuming `library != null`) because then we've somehow breached sane expectations.

Reply:
The other methods that take in an `ILibrary` are not part of an `IProvider` implementation.  They use the `ProviderId` property to retrieve the correct provider.  In that sense, there isn't a comparable situation - this is the only place where one could pass an `ILibrary` to the wrong `Provider`.

I know it's redundant, but it is a unique check (though similar) in each implementation.  I prefer to leave it as is.